### PR TITLE
Added implementation

### PR DESCRIPTION
**Pull request by oleksandrdziubiuk**

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: src/main/java/com/bankservice/bankservice/repository/RoleRepository.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.bankservice.bankservice.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("FROM roles r WHERE r.roleName =:roleName")
+    Optional<Role> getByRoleName(@Param("roleName") String roleName);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bankservice/bankservice/repository/RoleRepository.java]:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

File: src/main/java/com/bankservice/bankservice/repository/UserRepository.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.bankservice.bankservice.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("FROM users u JOIN FETCH u.roles WHERE u.phoneNumber =:phoneNumber")
+    Optional<User> getByPhoneNumber(@Param("phoneNumber") String phoneNumber);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bankservice/bankservice/repository/UserRepository.java]: in getById() method we should also join fetch roles

Comment (KseniiaMakarova) [inline src/main/java/com/bankservice/bankservice/repository/UserRepository.java]:
```suggestion
    @Query("FROM users u JOIN FETCH u.roles WHERE u.phoneNumber =:phoneNumber")
    Optional<User> getByPhoneNumber(String phoneNumber);
```
you can simplify this a bit

File: src/main/java/com/bankservice/bankservice/service/RoleService.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.bankservice.bankservice.service;
+
+import com.bankservice.bankservice.model.Role;
+import java.util.Optional;
+
+public interface RoleService {
+    Role save(Role role);
+
+    Optional<Role> getByName(String roleName);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bankservice/bankservice/service/RoleService.java]: let's return entity from service layer

File: src/main/java/com/bankservice/bankservice/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+public interface UserService {
+    User create(User user);
+
+    User update(User user);
+
+    void delete(Long id);
+
+    Optional<User> getById(Long id);
+
+    Optional<User> getByPhone(String phoneNumber);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bankservice/bankservice/service/UserService.java]: same

File: src/main/java/com/bankservice/bankservice/service/UserService.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.bankservice.bankservice.service;
+
+import com.bankservice.bankservice.model.User;
+import java.util.List;
+import java.util.Optional;
+
+public interface UserService {
+    User create(User user);
+
+    User update(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bankservice/bankservice/service/UserService.java]: please see checklist

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let’s configure access to h2 database through the console in `application.properties`

File: src/main/java/com/bankservice/bankservice/service/UserService.java [hunk 6]
```diff
@@ -2,18 +2,15 @@
 
 import com.bankservice.bankservice.model.User;
 import java.util.List;
-import java.util.Optional;
 
 public interface UserService {
     User create(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/bankservice/bankservice/service/UserService.java]: let's rename this method to `save()`

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
`application.properties` are still empty
you should be able to access your h2 DB from browser

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**